packages/clay-navigation/src/Breadcrumb.tsx

```diff
@@ -0,0 +1,122 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React from 'react';
+import warning from 'warning';
+import {BreadcrumbEllipsis} from './BreadcrumbEllipsis';
+import {BreadcrumbItem, IBreadcrumbItem} from './BreadcrumbItem';
+
+interface IProps extends React.HTMLAttributes<HTMLOListElement> {
+	ellipsisBuffer?: number;
+
+	ellipsisProps?: Object;
+
+	items: Array<IBreadcrumbItem>;
+
+	spritemap: string;
+}
+
+const getBufferList = (
+	items: Array<IBreadcrumbItem>,
+	start: number,
+	end: number,
+	ellipsisProps: Object,
+	spritemap: string
+): (IBreadcrumbItem | React.ReactElement)[] => {
+	const removedItems = items.slice(start, Math.max(end, start));
+
+	return removedItems.length > 1
+		? [
+				<BreadcrumbEllipsis
+					items={removedItems}
+					key="breadcrumbEllipsis"
+					spritemap={spritemap}
+					{...ellipsisProps}
+				/>,
+		  ]
+		: removedItems;
+};
+
+const getBreadcrumbItems = (
+	ellipsisBuffer: number,
+	items: Array<IBreadcrumbItem>,
+	ellipsisProps: Object,
+	spritemap: string
+) => {
+	const activeIndex = items.findIndex(item => item.active);
+
+	const lastIndex = items.length - 1;
+
+	const leftBufferEnd = activeIndex - ellipsisBuffer;
+
+	const rightBufferStart = activeIndex + ellipsisBuffer + 1;
+
+	const leftBuffer = getBufferList(
+		items,
+		1,
+		leftBufferEnd,
+		ellipsisProps,
+		spritemap
+	);
+	const rightBuffer = getBufferList(
+		items,
+		rightBufferStart,
+		lastIndex,
+		ellipsisProps,
+		spritemap
+	);
+
+	return [
+		items[0],
+		...leftBuffer,
+		...items.slice(
+			Math.max(activeIndex - ellipsisBuffer, 1),
+			Math.min(activeIndex + ellipsisBuffer + 1, lastIndex)
+		),
+		...rightBuffer,
+		items[lastIndex],
+	];
+};
+
+export const Breadcrumb: React.FunctionComponent<IProps> = ({
+	className,
+	ellipsisBuffer = 1,
+	ellipsisProps = {},
+	items,
+	spritemap,
+	...otherProps
+}: IProps) => {
+	warning(
+		items.findIndex(item => item.active) != -1,
+		'ClayNavigation.Breadcrumb expects at least one `active` item on `items`.'
+	);
+
+	return (
+		<ol {...otherProps} className={classNames('breadcrumb', className)}>
+			{(ellipsisBuffer
+				? getBreadcrumbItems(
+						ellipsisBuffer,
+						items,
+						ellipsisProps,
+						spritemap
+				  )
+				: items
+			).map((item: IBreadcrumbItem | React.ReactElement, i: number) =>
+				React.isValidElement(item) ? (
+					React.cloneElement(item, {key: `ellipsis${i}`})
+				) : (
+					<BreadcrumbItem
```

Comment:
What about `onClick` instead of href? I could imagine that would be used as well.

Reply:
I think we can keep both. But we have to think if keeping `href` is probably for cases that would have been a navigation using the browser and not a navigation through the React Router for example. One thought about this is:

- If you allow `href`, it may be cases where people will only use the component as if it were a taglib. Maybe I was wrong, but in v2 we added the href for those cases.
- If you allow `onClick`, it may be cases where you have control and want to do a navigation using the React Router for example.

So the question is, can we just allow `onClick` and force people to use components for full React applications rather than for a loose component, such as a taglib. But if you follow this path it can not be used in a taglib.

Reply:
For a react app with routing, I think the user would also just use an `href` and then let the touer handle changing the state. The more I think about it, the more `onClick` is likely used less and maybe wouldn't be used at all.

Reply:
So I guess im actually thinking the opposite, and only allowing href and keep them as anchor tags

Reply:
Oh no 😅, I'm not sure if `href` would help with the React Router because you need to use the `Link` component (at least that's how I used it 🙂).

So I would probably use `onClick` and then call `navigate` (I think this is Reach Router 🙂).

Reply:
Oh good point with `Link`. Maybe we need to add ability to pass in a custom component as well, since that is likely a important use case.

The reason I would want to avoid `onClick` only is because you wouldn't be able to right click open or hold cmd and click to open in a new tab. href is important for accessibility.

Reply:
> Oh good point with Link. Maybe we need to add ability to pass in a custom component as well, since that is likely a important use case.

Hmm, I had not thought about it, but I find it interesting.

> The reason I would want to avoid onClick only is because you wouldn't be able to right click open or hold cmd and click to open in a new tab. href is important for accessibility.

Yes you are right, we can keep both, I think both end up being important.

Reply:
Was added a property called `onItemClicked`